docs/source/tutorials/dev_bayesopt/surrogate_model.rst

```diff
@@ -0,0 +1,268 @@
+  ``update_params=False``, the current parameters are used without updating
+  them).
+* :class:`~syne_tune.optimizer.schedulers.searchers.bayesopt.tuning_algorithms.base_classes.Predictor`:
+  Wraps the posterior state computed by the ``Estimator``, allows for predictions.
+
+The fitting of surrogate models underlying a Bayesian optimization experiment
+happens in
+:class:`~syne_tune.optimizer.schedulers.searchers.bayesopt.models.model_transformer.ModelStateTransformer`,
+which interfaces between a model-based searcher and the surrogate model. The
+``ModelStateTransformer`` maintains the state of the experiment, where all data
+about observations and pending configurations are collected. Its
+:meth:`~syne_tune.optimizer.schedulers.searchers.bayesopt.models.model_transformer.ModelStateTransformer.fit`
+method triggers fitting the surrogate models to the current data (this step can
+be skipped for computational savings) and computing their posterior states.
+
+``ModelStateTransformer`` hands down these tasks to an object of type
+:class:`~syne_tune.optimizer.schedulers.searchers.bayesopt.models.estimator.Estimator`,
+which is specific to the surrogate model being used. For our Gaussian process
+example, this would be
+:class:`~syne_tune.optimizer.schedulers.searchers.bayesopt.models.gp_model.GaussProcEmpiricalBayesEstimator`.
+Here, parameters of the Gaussian process models (such as parameters of the
+covariance function) are fitted by marginal likelihood maximization, and the
+GP posterior state is computed.
+
+.. note::
+   To summarize, if your surrogate model needs to be fit to data, you need to
+   implement a subclass of
+   :class:`~syne_tune.optimizer.schedulers.searchers.bayesopt.models.estimator.Estimator`,
+   whose ``fit_from_state`` method takes in data in form of a
+   :class:`~syne_tune.optimizer.schedulers.searchers.bayesopt.datatypes.tuning_job_state.TuningJobState`
+   and returns a
+   :class:`~syne_tune.optimizer.schedulers.searchers.bayesopt.tuning_algorithms.base_classes.Predictor`.
+   You can use
+   :func:`~syne_tune.optimizer.schedulers.searchers.bayesopt.models.estimator.transform_state_to_data`
+   in order to convert the ``TuningJobState`` object into the usual pair of
+   feature matrix ``features`` and target vector ``targets``, along with
+   normalization of targets.
+
+Example
+-------
+
+Let us look at a simple example for how to implement a new surrogate model in
+Syne Tune, of the scikit-learn estimator based type. It does not come with some
+of the complexities of Gaussian process based surrogate models:
+
+* Fantasizing is not supported
+* MCMC (or ensemble predictions) is not supported
+* Gradient-based optimization of an acquisition function is not supported, in
+  that Bayesian optimization is scoring a finite number of candidates drawn
+  at random, selecting the best
+
+The full example code is given
+`here <../../examples.html#bayesian-optimization-with-scikit-learn-based-surrogate-model>`__.
```

Comment:
```
    def predict(self, X: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
        return self.ridge.predict(X, return_std=True)
```

this does not convert `X`. Is this done automatically? Is it always done automatically?

Reply:
``predict_candidates``. I insert something